### DISCUSSION TITLE
I need help for an emergency shutdown

### DISCUSSION THREAD

**Discussion by umtauscher** [echavet/MitsubishiCN105ESPHome#127] [General]
Hi,
since I am a total ESPHome noob, I wanted to ask for some help with enhancing the electronics.
Due to a flood that was caused by a blocked condens water pipe, I would like to install a safety feature that shuts down the heatpump when that happens again.
I found a neat little optical sensor that I can screw into the tank. https://www.eptsensor.com/level-sensor/single-point-level-sensor/m12-thread-arduino-water-level-sensor.html

I put this into the config file:
binary_sensor:
  - platform: gpio
    name: "Kondenswasser"
    pin: GPIO16
    filters:
      - invert:
     
and it works well, but I have no idea how to switch off the heatpump when the sensor gets aktive. Naturally I can put an automation into homeassistant that does that, but I rather want the heatpunp to do this on itsself.
I would apreciate any help.
Thanks
Wilhelm

**Comment by umtauscher** [comment]
hmm, is it so difficult or did just nowbody read my question?

**Comment by echavet** [comment]
hello @humtauscher,
sorry i didn’t reply

this is not a feature I plan to develop because you can easily use lambda
in esphome yaml that triggers on state change of a home assistant sensor

this looks like a basic esphome feature which is not in the scope of this
firmware.




Envoyé avec Gmail Mobile


Le mar. 16 juil. 2024 à 16:26, umtauscher ***@***.***> a
écrit :

**Comment by umtauscher** [comment]
thanks for answering. I just wanted to know, if there is any way the esp could shutdown the system by itsself, when it detects the drainage is blocket. The sensor works perfectly, I just wanted to make the shutdown autonomous. 
As you can see, I modified the config to support the sensor. Is there any way I could implement an action that shuts the system of in such a case?

**Reply by umtauscher** [reply]
I tried to read the lambda part of the esphome docs but it could as well be in chinese.
Those examples make no sense to me whatsoever.

**Comment by phidauex** [comment]
You probably aren't getting full answers here because the feature you are looking for is a generic ESPHome feature, not something unique to this firmware. Using ESPHome functions you can do an amazing number of different things, and bringing it up in an ESPHome forum would probably get you more answers.

To point you in the right direction, you want to set something up using Actions. https://esphome.io/automations/actions#triggers

I have no idea if this example will work, but from a glance at the documentation it is at least headed in the right direction and should give you some things to experiment with. This uses the on_state automation trigger of your binary sensor (https://esphome.io/components/binary_sensor/#binary-sensor-on-state) to trigger the climate.control action from the Climate integration (https://esphome.io/components/climate/#climate-control-action).

The indentation may be a bit off here, and the on_state control may be inverted (IE this might turn the heatpump off when your sensor is NOT triggered), but hopefully it gets you going.

```yaml
binary_sensor:
  - platform: gpio
    name: "Kondenswasser"
    pin: GPIO16
    filters:
      - invert:
    on_state:
      then:
        - climate.control:
            id: hp
            mode: OFF
```

**Comment by umtauscher** [comment]
Thank you very much, I will try that and report back.
Cheers
